packages/blocks/src/api/raw-handling/test/special-comment-converter.js

```diff
@@ -49,6 +49,44 @@ describe( 'specialCommentConverter', () => {
 			<p>Third paragraph</p>`
 		);
 	} );
+	describe( 'when more comment is inside paragraph', () => {
+		it( 'should split the paragraph', () => {
+			const output = deepFilterHTML(
+				`<p>First part<!--more-->second part</p>`,
+				[ specialCommentConverter ]
+			);
+			expect( output ).toEqual(
+				`<p>First part</p><wp-block data-block=\"core/more\"></wp-block><p>second part</p>`
+			);
+		} );
+		it( 'should handle complex cases', () => {
```

Comment:
Minor suggestion for a more descriptive name:

```suggestion
		it( 'should preserve inline formatting', () => {
```